java-surefire/src/main/java/org/sonar/plugins/surefire/api/SurefireUtils.java

```diff
@@ -47,27 +61,61 @@ private SurefireUtils() {
    * @param pathResolver Path solver.
    * @return The directory containing the surefire reports or default one (target/surefire-reports) if not found (not configured or not found).
    */
-  public static File getReportsDirectory(Settings settings, FileSystem fs, PathResolver pathResolver) {
+  public static List<File> getReportsDirectories(Settings settings, FileSystem fs, PathResolver pathResolver) {
     File dir = getReportsDirectoryFromProperty(settings, fs, pathResolver);
+    List<File> dirs = getReportsDirectoriesFromProperty(settings, fs, pathResolver);
+    if (dir != null ^ dirs != null) {
```

Comment:
I find this quite complicated (I don't like `^` operator :-) ) , I think I have the same logic with one less return and without `^` operator.
```
    if (dirs != null) {
      if (dir != null) {
        // both properties are set
        LOGGER.info("Property '" + SUREFIRE_REPORTS_PATH_PROPERTY + "' is deprecated and will be ignored, as property '" + SUREFIRE_REPORT_PATHS_PROPERTY + "' is also set.");
      }
      return dirs;
    }
    if (dir != null) {
      return Collections.singletonList(dir);
    }
    // both properties are not set
    return Collections.singletonList(new File(fs.baseDir(), "target/surefire-reports"));;
```